backend/service/topology/topology.go

```diff
@@ -0,0 +1,54 @@
+package topology
+
+import (
+	"database/sql"
+	"errors"
+
+	"github.com/golang/protobuf/ptypes"
+	"github.com/golang/protobuf/ptypes/any"
+	"github.com/uber-go/tally"
+	"go.uber.org/zap"
+
+	topologyv1 "github.com/lyft/clutch/backend/api/config/service/topology/v1"
+	"github.com/lyft/clutch/backend/service"
+	pgservice "github.com/lyft/clutch/backend/service/db/postgres"
+)
+
+const Name = "clutch.service.topology"
+
+type Service interface{}
+
+type client struct {
+	config *topologyv1.Config
+
+	db    *sql.DB
+	log   *zap.Logger
+	scope tally.Scope
+}
+
+func New(cfg *any.Any, logger *zap.Logger, scope tally.Scope) (service.Service, error) {
+	topologyConfig := &topologyv1.Config{}
+	err := ptypes.UnmarshalAny(cfg, topologyConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	p, ok := service.Registry[pgservice.Name]
+	if !ok {
+		return nil, errors.New("Please config the datastore [clutch.service.db.postgres] to use the topology service")
+	}
+
+	dbClient, ok := p.(pgservice.Client)
+	if !ok {
+		return nil, errors.New("Unable to get the datastore client")
+	}
+
+	c, err := &client{
```

Comment:
can you just return here?

Reply:
Ah yes I could have although I had some additional code after the client was setup in my draft [pr](https://github.com/lyft/clutch/pull/362). I'll clean this up for now.